exec-engine.md

```diff
@@ -0,0 +1,146 @@
+# L2 Execution Engine
+
+The rollup mirrors the layer abstraction that the Merge introduces to Ethereum L1:
+- Consensus: *Determine* canonical chain, propagate *blocks*, maintain *system-state*
+- Execution: *Process* canonical chain, propagate *transactions*, maintain *user-state*
+
+The consensus task is implemented by the rollup-node, deriving most functionality from L1.
+The execution task is implemented by the engine, providing an equivalent user environment as L1, with deposits.
```

Comment:
We can remove this from this doc, but I thought we should define this consensus-execution separation somewhere, as context for this document. Suggestions for intro-text welcome.

Also, regarding introduction: we should automate TOC generation for spec docs.